api/custom_auth/serializers.py

```diff
@@ -17,6 +18,23 @@
 )
 
 
+class CustomTokenCreateSerializer(TokenCreateSerializer):
+    def validate(self, attrs):
```

Comment:
We seem to have missed [these lines](https://github.com/sunscrapers/djoser/blob/master/djoser/serializers.py#L124-L127) in overriding this function, was that intentional?

Reply:
Wouldn't a better way to do this be something like the following?  

```python
    def validate(self, attrs):
        if settings.LOGIN_FIELD != FFAdminUser.USERNAME_FIELD:
            attrs[settings.LOGIN_FIELD] = attrs.pop(FFAdminUser.USERNAME_FIELD]
        return super().validate(attrs)
```

Maybe we can even be more specific and check for LDAP being installed or something? This serializer could also technically live in the `flagsmith-ldap` module and we just update `REST_FRAMEWORK["SERIALIZERS"]["token_create"]` if LDAP is installed (although this would mean having to add a dependency on djoser to `flagsmith-ldap` I guess).

The main reason that I prefer the approach above though is that it means we don't have to worry if we upgrade djoser in the future for example as we're still calling the super validate method.

Reply:
I agree with everything except the statement, "This serializer could also technically live in the flagsmith-ldap module." While I agree that we only have to do this because of LDAP, I don't think it's something that LDAP should concern itself with. It should be the responsibility of the application to match the interface of the library (LDAP) and not the other way around.

Reply:
> We seem to have missed [these lines](https://github.com/sunscrapers/djoser/blob/master/djoser/serializers.py#L124-L127) in overriding this function, was that intentional?

Yep, I don't think that part is relevant for us

Reply:
Yep, fair enough 👍

Reply:
> Wouldn't a better way to do this be something like the following?

No, that won't work because of [this line](https://github.com/sunscrapers/djoser/blob/master/djoser/serializers.py#L116) if you change `settings.LOGIN_FIELD`(in djoser settings) that serializer will just ignore the email field 
ps: In the current implementation, we are overriding djoser settings instead we are updating our own `settings`

Reply:
> The main reason that I prefer the approach above though is that it means we don't have to worry if we upgrade djoser in the future for example as we're still calling the super validate method.

I think we still have to, e.g: if they change validate method? And we are overriding djoser at a lot of places, so we would absolutely have to check every update

Reply:
Ok, so just override __init__ as well?

Reply:
> Ok, so just override `__init__` as well?

Then we pretty much loose all of this? 

```
The main reason that I prefer the approach above though is that it means we don't have to worry if we upgrade djoser in the future for example as we're still calling the super validate method.
```

Reply:
Well, no, we can just call super?

Reply:
But, there is not much value left in doing that anymore?

Reply:
I don't really see how [this](https://github.com/Flagsmith/flagsmith/compare/fix/ldap-login...fix/ldap-login-v2) isn't better than what we have here?

Yes, djoser changes may cause issues in the future, but I think we'd know about them via our tests vs just re-implementing the code as we have here. In this implementation, I'm worried that e.g. security changes would slip through the net.